mysqld_exporter.go

```diff
@@ -737,6 +742,14 @@ func (e *Exporter) scrape(ch chan<- prometheus.Metric) {
 	}
 	defer db.Close()
 
+	if *slowLogFilter {
+		sessionSettingsRows, err := db.Query(sessionSettingsQuery)
+		if err != nil {
+			return err
+		}
+		defer sessionSettingsRows.Close()
```

Comment:
Is there a reason to defer the Close call to the end of the function? sessionSettingsRows goes out of scope here, so you could close it right ahead.